incubator/exporter-stdout/src/main/java/io/aklivity/zilla/runtime/exporter/stdout/internal/spy/OneToOneRingBufferSpy.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2021-2023 Aklivity Inc
+ *
+ * Licensed under the Aklivity Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.exporter.stdout.internal.spy;
+
+import static org.agrona.BitUtil.align;
+import static org.agrona.concurrent.ringbuffer.RecordDescriptor.ALIGNMENT;
+import static org.agrona.concurrent.ringbuffer.RecordDescriptor.HEADER_LENGTH;
+import static org.agrona.concurrent.ringbuffer.RecordDescriptor.lengthOffset;
+import static org.agrona.concurrent.ringbuffer.RecordDescriptor.typeOffset;
+import static org.agrona.concurrent.ringbuffer.RingBuffer.PADDING_MSG_TYPE_ID;
+import static org.agrona.concurrent.ringbuffer.RingBufferDescriptor.HEAD_POSITION_OFFSET;
+import static org.agrona.concurrent.ringbuffer.RingBufferDescriptor.TAIL_POSITION_OFFSET;
+import static org.agrona.concurrent.ringbuffer.RingBufferDescriptor.TRAILER_LENGTH;
+import static org.agrona.concurrent.ringbuffer.RingBufferDescriptor.checkCapacity;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.agrona.DirectBuffer;
+import org.agrona.concurrent.AtomicBuffer;
+
+import io.aklivity.zilla.runtime.engine.binding.function.MessagePredicate;
+
+public class OneToOneRingBufferSpy implements RingBufferSpy
```

Comment:
This package should live in the engine, and an exporter should be able to ask the engine to create such a spy to read events.